proxy.go

```diff
@@ -0,0 +1,32 @@
+package proxy
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/ops"
+)
+
+var (
+	log = golog.LoggerFor("proxy")
+)
+
+type DialFunc func(network, addr string) (conn net.Conn, err error)
+
+type Interceptor func(op ops.Op, w http.ResponseWriter, req *http.Request)
```

Comment:
It feels to me it's unnecessary to introduce dependency to `ops` package which is more suitable at the application level. How about `func(w http.ResponseWriter, req *http.Request) error`, and let the caller decide what to do with the error?